Tests/VaporTests/AsyncRequestTests.swift

```diff
@@ -195,6 +195,42 @@ final class AsyncRequestTests: XCTestCase {
         requestHandlerTask.withLockedValue { $0?.cancel() }
         try await httpClient.shutdown()
     }
+
+    // https://github.com/vapor/vapor/issues/2985
+    func testLargeBodyCollectionDoesntCrash() async throws {
+        app.http.server.configuration.hostname = "127.0.0.1"
+        app.http.server.configuration.port = 0
+
+        app.on(.POST, "upload", body: .stream, use: { request async throws -> String  in
+            let buffer = try await request.body.collect(upTo: Int.max)
+            return "Received \(buffer.readableBytes) bytes"
+        })
+
+        app.environment.arguments = ["serve"]
+        try await app.startup()
+
+        XCTAssertNotNil(app.http.server.shared.localAddress)
+        guard 
+            let localAddress = app.http.server.shared.localAddress,
+            let ip = localAddress.ipAddress,
+            let port = localAddress.port 
+        else {
+            XCTFail("couldn't get ip/port from \(app.http.server.shared.localAddress.debugDescription)")
+            return
+        }
+
+        let fiftyMB = ByteBuffer(repeating: 0x41, count: 600 * 1024 * 1024)
```

Comment:
Seems like you changed it to 600MB
```suggestion
        let bigBody = ByteBuffer(repeating: 0x41, count: 600 * 1024 * 1024)
```

Reply:
Oops, thanks!